docs/qml/qaoa.md

```diff
@@ -34,57 +36,111 @@ The goal of the MaxCut algorithm is to maximize the following cost function:
 
 $$\mathcal{C}(p) = \sum_{\alpha}^m \mathcal{C}_{\alpha}(p)$$
 
-where $p$ is the given partition of the graph, $\alpha$ is an index over the edges and $\mathcal{C}_{\alpha}(p)$ is written
+where $p$ is a given cut of the graph, $\alpha$ is an index over the edges and $\mathcal{C}_{\alpha}(p)$ is written
 such that if the nodes connected by the $\alpha$ edge are in the same set, it returns $0$, otherwise it returns $1$.
 
+Since in this tutorial we are only dealing with small graphs, we can find the maximum cut by brute force to make sure QAOA works as intended. Recall that every cut is represented by two bitstrings, e.g. cuts "0011" and "1100" are equivalent as they represent the same cut.
+
+```python exec="on" source="material-block" result="json" session="qaoa"
+# Calculate the maximum cut and the associated maximum cost
+def calculate_cost(bitstring: str, adjacency_matrix: np.ndarray) -> float:
+    """Returns the cost of a given cut (represented by the bitstring)"""
+    n = len(bitstring)
+    cost = 0
+    for i in range(n):
+        for j in range(i + 1, n):
+            if bitstring[i] != bitstring[j]:
+                cost += adjacency_matrix[i, j]
+    return cost
+
+
+all_possible_cuts = [bin(k)[2:].rjust(n_nodes, "0") for k in range(2**n_nodes)]
+adjacency_matrix = nx.adjacency_matrix(graph)
+all_costs = [calculate_cost(cut, adjacency_matrix) for cut in all_possible_cuts]
+maxcost = max(all_costs)
+maxcut = all_possible_cuts[all_costs.index(maxcost)]
+other_maxcut_bs = "".join(str((int(i) + 1) % 2) for i in maxcut)
+print(
+    f"The maximum cut is represented by the bitstrings {maxcut} and {other_maxcut_bs}, with a cost of {maxcost}"
+)
```

Comment:
This code is a bit hard to read. Could you add some description text on what these lines are doing?